cli/command/container/run.go

```diff
@@ -209,7 +205,7 @@ func runRun(dockerCli *command.DockerCli, flags *pflag.FlagSet, opts *runOptions
 		})
 	}
 
-	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, hostConfig.AutoRemove)
+	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
```

Comment:
hostConfig.AutoRemove will be overridden in `createContainer()` if API 1.24 is used. Even though that shouldn't affect the code here, I thought it was closer to the truth to look at the `--rm` flag, not the container config here.